aws/table_aws_workspaces_workspace.go

```diff
@@ -194,11 +201,15 @@ func getWorkspace(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateDat
 	// Get call
 	data, err := svc.DescribeWorkspaces(params)
 	if err != nil {
+		// AWS workspaces is not available in every region yet. This section of code handles the errors that we get when the API call tries to use unsupported regions endpoint (it throws "no such host" error message)
+		if strings.Contains(err.Error(), "no such host") {
+			return nil, nil
+		}
 		plugin.Logger(ctx).Error("DescribeWorkspaces", "ERROR", err)
 		return nil, err
 	}
 
-	if data.Workspaces != nil {
+	if data.Workspaces != nil && len(data.Workspaces) > 0 {
```

Comment:
Do we need to check for both conditions? What are the possible responses we get back from the Workspaces API?

Reply:
len(data.Workspaces) > 0 is actually the required one. can remove the 1st one. It may generate error - `rpc error: code = Internal desc = hydrate function getWorkspace failed with panic runtime error: index out of range [0] with length 0`